syntax = "proto3";
package RevaProtocol;
option java_package = "reva.protocol";

service RevaToolSymbolService {
  rpc GetSymbols(RevaGetSymbolsRequest) returns (RevaGetSymbolsResponse);
  rpc SetSymbolName(RevaSetSymbolNameRequest) returns (RevaSetSymbolNameResponse);
  rpc GetSymbol(RevaSymbolRequest) returns (RevaSymbolResponse);
}

service RevaLLMSymbolService {
  rpc GetNewSymbolName(RevaGetNewSymbolNameRequest) returns (RevaGetNewSymbolNameResponse);
}

message RevaGetSymbolsRequest {
}

enum SymbolType {
  FUNCTION = 0;
  DATA = 1;
  LABEL = 2;
}

message RevaSymbolRequest {
  string address = 1;
  string name = 2;
}

message RevaSymbolResponse {
  string name = 1;
  string address = 2;
  SymbolType type = 3;
}

message RevaGetSymbolsResponse {
  repeated string symbols = 1;
}

message RevaSetSymbolNameRequest {
  string old_name_or_address = 1;
  string new_name = 2;
}

message RevaGetNewSymbolNameRequest {
  string symbol_name = 1;
}

message RevaGetNewSymbolNameResponse {
  string new_symbol_name = 1;
}

message RevaSetSymbolNameResponse {
}